### ISSUE TITLE
gapi.auth.authorize with immediate=true fails on Internet Explorer 11 with default Internet Zone settings for https://accounts.google.com

### ISSUE THREAD

**Issue by wonderfly** [google/google-api-javascript-client#146]
_From [mleather...@gmail.com](https://code.google.com/u/103165304670922340726/) on May 06, 2014 12:31:57_

Using Internet Explorer 11, go to " http://royalpvpgod.com/t0.html " (the source of this web page is enclosed).

Hit Authorize (immediate=false). It will bring up a popup window and then print "SUCCESS" in the text window. 

Now hit Authorize (immediate=true). This will print "FAILURE". According to the documentation, so long as the user is logged in this should do a "behind the scenes" token refresh and return successfully.

This was definitely working 2 days ago on IE. It appears to have broken around 5pm PST last Sunday. It still works fine on Firefox, and Chrome (haven't tried the others).

**Attachment:** [t0.html](http://code.google.com/p/google-api-javascript-client/issues/detail?id=146)

_Original issue: http://code.google.com/p/google-api-javascript-client/issues/detail?id=146_

**Comment by bsittler** [comment]
Unfortunately some history appears to have been lost in the issue tracker transition to github. There was a developer report in the Google Code tracker indicating that the issue has been resolved. Is anyone still seeing the incorrect behavior?

**Comment by ovoytovich** [comment]
I don't see this error on IE8 and IE9, but currently for IE11 whole OAuth flow fails for me. In the pop-up there's nothing displayed where there should be displayed scopes that are authorized.

**Comment by bloolizard** [comment]
hoping this issue will be resolved.  affecting my users!

**Comment by morsanu** [comment]
Any news on this one?

**Comment by tjoneseng** [comment]
I abandoned the Google SDK in favor of hello.js.  That seems to work great across all browsers.

**Comment by bsittler** [comment]
Is anyone still seeing this problem? If so, do the mentioned workarounds resolve the issue? If the issue is in fact the same as the one for which this ticket was originally opened, it is due to a design decision in Internet Explorer combined with Zone security settings which disallow communication between the client application and Google's servers, and a fix in the google-api-javascript-client may be infeasible.

**Comment by hakimCZE** [comment]
Yes, I still see this problem. In Edge, I see "redirecting" white page. In IE 11 there is page "page doesnt exist"

**Comment by bsittler** [comment]
@hakimCZE is the app on which you see this behavior in the same security zone (likely Internet Zone) as https://accounts.google.com ? Also, what is the URL in the address bar of the "page doesn't exist" window?

**Comment by TMSCH** [comment]
@hakimCZE are you using gapi.auth.authorize or gapi.auth2.authorize?

**Comment by sachinyadav1990** [comment]
yeah this issue is coming now.. trusted site is not practical solution for this. Please advice something else...?

**Comment by hakimCZE** [comment]
@TMSCH we are using gapi.auth2.authorize
@bsittler and yes, the authorization page is on live public site. The second I am not sure, as there is redirect from our internal devel site to OAuth login page. Developer that has responsibility for it is playing dead and is unwilling to do something about it, so I cant provide more info right now :/

**Comment by bsittler** [comment]
@sachinyadav1990 are both accounts.google.com and the affected web application in the same Internet Security Zone?

**Comment by dvanderb** [comment]
This looks like a question about Google Login and Authentication. Please address this question to the google-signin tag on stackoverflow(https://stackoverflow.com/questions/tagged/google-signin)

If you believe this is in error, please reopen the issue.
